hilti/toolchain/include/ast/operator-registry.h

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2020-2023 by the Zeek Project. See LICENSE for details.
+
+#pragma once
+
+#include <list>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <hilti/ast/operator.h>
+
+#define HILTI_OPERATOR(ns, cls)                                                                                        \
+    Result<hilti::ResolvedOperatorPtr> instantiate(hilti::Builder* builder, Expressions operands, const Meta& meta)    \
+        const final {                                                                                                  \
+        return {ns::operator_::cls::create(builder->context(), this, result(builder, operands, meta),                  \
+                                           std::move(operands), meta)};                                                \
+    }                                                                                                                  \
+                                                                                                                       \
+    std::string name() const final { return hilti::util::replace(#cls, "_::", "::"); }                                 \
+    std::string _typename() const final { return hilti::util::typename_(*this); }
+
+#define HILTI_OPERATOR_IMPLEMENTATION(cls)                                                                             \
+    namespace {                                                                                                        \
+    static hilti::operator_::Register<cls> _operator_##cls;                                                            \
+    }
+
+namespace hilti::operator_ {
+
+/** Singleton registering available operators. */
+class Registry {
+public:
+    /**
+     * Returns all available built-on operators of kind function call matching
+     * a given function name.
+     */
+    const auto& byBuiltinFunctionID(const ID& id) { return _operators_by_builtin_function[id]; }
+
```

Comment:
Yeah, that's what we want. But I don't see what needs changing, it doesn't matter to the caller.

Reply:
> But I don't see what needs changing, it doesn't matter to the caller.
A caller wouldn't be able to e.g., invoke this function on a `const Registry&`.

As for get-or-insert-default semantics, they also cannot mutate the default-inserted element since this function returns a `const` ref and the types simply do not allow it. That probably lead me to believe that this was a simple getter (which I'd expected to be `const`).